lib/block_view.go

```diff
@@ -200,6 +203,78 @@ type LikeEntry struct {
 	isDeleted bool
 }
 
+func MakeNFTKey(nftPostHash *BlockHash, serialNumber uint64) NFTKey {
+	return NFTKey{
+		NFTPostHash:  *nftPostHash,
+		SerialNumber: serialNumber,
+	}
+}
+
+type NFTKey struct {
+	NFTPostHash  BlockHash
+	SerialNumber uint64
+}
+
+// This struct defines an individual NFT owned by a PKID. An NFT entry  maps to a single
+// postEntry, but a single postEntry can map to multiple NFT entries. Each NFT copy is
+// defined by a serial number, which denotes it's place in the set (ie. #1 of 100).
+type NFTEntry struct {
+	LastOwnerPKID              *PKID // This is needed to decrypt unlockable text.
+	OwnerPKID                  *PKID
+	NFTPostHash                *BlockHash
+	SerialNumber               uint64
+	IsForSale                  bool
+	MinBidAmountNanos          uint64
+	UnlockableText             []byte
+	LastAcceptedBidAmountNanos uint64
+
+	// Whether or not this entry is deleted in the view.
+	isDeleted bool
+}
+
+func MakeNFTOwnershipKey(ownerPKID *PKID, isForSale bool, lastAcceptedBidAmountNanos uint64, nftPostHash *BlockHash, serialNumber uint64) NFTOwnershipKey {
+	return NFTOwnershipKey{
+		OwnerPKID:                  *ownerPKID,
+		IsForSale:                  isForSale,
+		LastAcceptedBidAmountNanos: lastAcceptedBidAmountNanos,
+		NFTPostHash:                *nftPostHash,
+		SerialNumber:               serialNumber,
+	}
+}
+
+type NFTOwnershipKey struct {
```

Comment:
Add a comment on how this is used. Generally good to add comments on all new structs.

Reply:
Seems like the only field that's here that's not on the NFTEntry is the LastAcceptedBidAmountNanos. Couldn't we just lump that into the NFTEntry?

Reply:
Conclusion from conversation is that we should try and delete this, along with the corresponding flush logic. The DBDeleteNFTMappings function can be updated to this:

```
func DBDeleteNFTMappingsWithTxn(txn *badger.Txn, nftPostHash *BlockHash, serialNumber uint64) error {

	// First pull up the mapping that exists for the post / serial # passed in.
	// If one doesn't exist then there's nothing to do.
	nftEntry := DBGetNFTEntryByPostHashSerialNumberWithTxn(txn, nftPostHash, serialNumber)
	if nftEntry == nil {
		return nil
	}

	// When an nftEntry exists, delete the mapping.
	if err := txn.Delete(_dbKeyForPublicKeyIsForSaleBidAmountNanosNFTPostHashSerialNumber(
		nftEntry.OwnerPKID, nftEntry.IsForSale, nftEntry.MinBidAmountNanos, nftPostHash, serialNumber)); err != nil {
		return errors.Wrapf(err, "DbDeleteNFTMappingsWithTxn: Deleting "+
			"nft mapping for pkid %v post hash %v serial number %d", nftEntry.OwnerPKID, nftPostHash, serialNumber)
	}

	// When an nftEntry exists, delete the mapping.
	if err := txn.Delete(_dbKeyForNFTPostHashSerialNumber(nftPostHash, serialNumber)); err != nil {
		return errors.Wrapf(err, "DbDeleteNFTMappingsWithTxn: Deleting "+
			"nft mapping for post hash %v serial number %d", nftPostHash, serialNumber)
	}

	return nil
}
```